verifier/src/x509.rs

```diff
@@ -82,7 +98,11 @@ impl UnverifiedCertChain {
     ///
     /// # Errors
     /// `Error::MbedTls` if the certificate chain is not valid.
-    pub fn verify(self, trust_anchor: &TrustAnchor) -> Result<VerifiedCertChain> {
+    pub fn verify(
+        self,
+        trust_anchor: &TrustAnchor,
+        mut crl: CertificateRevocationList,
```

Comment:
Not fond of the "mut crl" but that's the `verify_with_profile` interface and there isn't a clone available for the underlying CRL type. I decided to consume the crl here instead of leaving it in an unknown mutated state.